scarlet/operator.py

```diff
@@ -597,17 +570,20 @@ def getRadialMonotonicWeights(shape, useNearest=True, minGradient=1, center=None
         cosWeight[invalidPix] = 0
         cosWeight[mask] = 0
 
-        if useNearest:
+        if neighbor_weight == "nearest":
             # Only use a single pixel most in line with peak
             cosNorm = np.zeros_like(cosWeight)
             columnIndices = np.arange(cosWeight.shape[1])
             maxIndices = np.argmax(cosWeight, axis=0)
             indices = maxIndices * cosNorm.shape[1] + columnIndices
             indices = np.unravel_index(indices, cosNorm.shape)
-            cosNorm[indices] = minGradient
+            cosNorm[indices] = 1
             # Remove the reference for the peak pixel
             cosNorm[:, px + py * shape[1]] = 0
         else:
+            if neighbor_weight == "flat":
+                cosWeight[cosWeight != 0] = 1
+
```

Comment:
Overall I like this except that since "flat" is the default there is no reason to go through all of the trig to build the `cosWeight` version of `cosNorm`. TBH I'm not sure why we were doing this every time before. So I think it would be faster to only calculate the angles and `cosWeight` if `neighbor_weight=="angles"`. For flat weights you can just check that `dx**2 + dy**2 < x**2+y**2`, where `x` and `y` are the coordinates of each pixel.